OloEditor/src/Panels/SceneHierarchyPanel.cpp

```
@@ -44,116 +52,226 @@ namespace OloEngine
+                                { m_SelectionContext = {}; }));
+                        }
+                        else
+                        {
+                            auto canvas = m_Context->CreateEntity("UI Canvas");
+                            canvas.AddComponent<UICanvasComponent>();
+                            canvas.AddComponent<UIRectTransformComponent>();
+                            m_SelectionContext = canvas;
+                        }
+                    };
+
+                    auto createUIWidgetWithUndo = [this](const char* name, auto addComponentFn)
```

Clang (14.0.6)

[note] nesting level increased to 4

(clang)